src/App.tsx

```diff
@@ -1,39 +1,44 @@
-import React from 'react';
+import { Component } from 'react';
+// import { render } from 'react-dom';
 import './App.scss';
+import { Clock } from './Clock';
 
-function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
-
-  return `Clock-${value}`;
+interface State {
+  hasClock: boolean;
+  name: string;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
-
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    name: 'Clock-0',
+  };
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.handleRihgtClick);
+    document.addEventListener('click', this.handleLeftClick);
+  }
+
+  componentWillUnmount(): void {
```

Comment:
It's extra, bc TypeScript already knows about React types
Also, try to keep one code styles
```suggestion
  componentWillUnmount() {
```